src/NHibernate/Linq/Visitors/ExpressionKeyVisitor.cs

```diff
@@ -163,20 +163,21 @@ protected override Expression VisitMethodCall(MethodCallExpression expression)
 		{
 			var old = insideSelectClause;
 
-			switch (expression.Method.Name)
-			{
-				case "First":
-				case "FirstOrDefault":
-				case "Single":
-				case "SingleOrDefault":
-				case "Select":
-				case "GroupBy":
-					insideSelectClause = true;
-					break;
-				default:
-					insideSelectClause = false;
-					break;
-			}
+			if (expression.Method.DeclaringType?.Namespace == "System.Linq")
```

Comment:
It should have `esle` branch.

Reply:
For putting what in it? All the point of that `if` is to not alter `insideSelectClause`, considering that only System.Linq calls may represent an exit out of select clause.

Reply:
To put `insideSelectClause = false;` as it is in the `default` branch of switch statement.

Reply:
Ok, I've checked that my suggestion does not work.